### PR TITLE
:seedling: build(deps): bump github.com/operator-framework/operator-registry from v1.41 to v1.43 and  github.com/operator-framework/api from v0.23.0 to v0.25.0

### PR DESCRIPTION
**Pull request by perdasilva**
**Description of the change:**

**Motivation for the change:**

**Architectural changes:**

**Testing remarks:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Bug fixes are accompanied by regression test(s)
- [ ] e2e tests and flake fixes are accompanied evidence of flake testing, e.g. executing the test 100(0) times
- [ ] tech debt/todo is accompanied by issue link(s) in comments in the surrounding code
- [ ] Tests are comprehensible, e.g. Ginkgo DSL is being used appropriately
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive
- [ ] Tests marked as `[FLAKE]` are truly flaky and have an issue
- [ ] Code is properly formatted

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/operator-framework/operator-lifecycle-manager
 
-go 1.22
```
[the remark this answers is not in the document]
Comment (perdasilva) [reply go.mod:3]: go mod vendor did that...I'm on 1.22....any ideas?

[the remark this answers is not in the document]
Comment (dinhxuanvu) [reply go.mod:3]: This is just go tooling chance that kind of started in go 1.21. It shouldn't matter here given you have the line `go 1.22.0` below.

**Comment by dinhxuanvu** [comment]
This PR needs a round of `codegen` run to update the CRDs.

**Comment by perdasilva** [comment]
Yeah, we need [this](https://github.com/operator-framework/operator-lifecycle-manager/pull/3240) first

**Comment by dinhxuanvu** [comment]
> Yeah, we need [this](https://github.com/operator-framework/operator-lifecycle-manager/pull/3240) first

You guys should plan a proper k8s rebase flow across all components to ensure a smooth operation and perhaps document it. k8s minor version bumps may come with some required changes. If it comes with a go version bump like this one (k8s 1.30 requires go 1.22), it may need even more changes.

**Comment by perdasilva** [comment]
> You guys should plan a proper k8s rebase flow across all components to ensure a smooth operation and perhaps document it. k8s minor version bumps may come with some required changes. If it comes with a go version bump like this one (k8s 1.30 requires go 1.22), it may need even more changes.

I think this basically brings the trinity (api/registry/olm) to go 1.22 and k8s 1.30.
I had to do a lot of tinkering around the codegen bits because the scripts we were using were long depracated (and got removed in 1.30 XD)

**Comment by perdasilva** [comment]
looks like kube is picking up a couple of api violations:

```
CustomResourceDefinition.apiextensions.k8s.io "clusterserviceversions.operators.coreos.com" is invalid: [spec.validation.openAPIV3Schema.properties[spec].properties[install].properties[spec].properties[deployments].items.properties[spec].properties[template].properties[spec].properties[imagePullSecrets].items.properties[name].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[install].properties[spec].properties[deployments].items.properties[spec].properties[template].properties[spec].properties[hostAliases].items.properties[ip].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property]
```

Maybe we need to tune generation? or fix the apis?

**Comment by dinhxuanvu** [comment]
> looks like kube is picking up a couple of api violations:
> 
> ```
> CustomResourceDefinition.apiextensions.k8s.io "clusterserviceversions.operators.coreos.com" is invalid: [spec.validation.openAPIV3Schema.properties[spec].properties[install].properties[spec].properties[deployments].items.properties[spec].properties[template].properties[spec].properties[imagePullSecrets].items.properties[name].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[install].properties[spec].properties[deployments].items.properties[spec].properties[template].properties[spec].properties[hostAliases].items.properties[ip].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property]
> ```
> 
> Maybe we need to tune generation? or fix the apis?

This CRD has been generated years ago and I don’t remember when was the last it got a proper update. You may need to go back to the code where the api is constructed and  check the comments to see if those k8s tags are still good.

**Comment by perdasilva** [comment]
> > looks like kube is picking up a couple of api violations:
> > ```
> > CustomResourceDefinition.apiextensions.k8s.io "clusterserviceversions.operators.coreos.com" is invalid: [spec.validation.openAPIV3Schema.properties[spec].properties[install].properties[spec].properties[deployments].items.properties[spec].properties[template].properties[spec].properties[imagePullSecrets].items.properties[name].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[install].properties[spec].properties[deployments].items.properties[spec].properties[template].properties[spec].properties[hostAliases].items.properties[ip].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Maybe we need to tune generation? or fix the apis?
> 
> This CRD has been generated years ago and I don’t remember when was the last it got a proper update. You may need to go back to the code where the api is constructed and check the comments to see if those k8s tags are still good.

Bad k8s version - seems running on v1.30+ fixes the issue

**Comment by perdasilva** [comment]
closing in favor of #3282 - which solves the issue by pinning a couple of breaking dependencies

**Pull request closed without merging**